// @clerk/shared/dist/authorization-errors.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/authorization-errors.ts
var authorization_errors_exports = {};
__export(authorization_errors_exports, {
  __experimental_isReverificationHint: () => __experimental_isReverificationHint,
  __experimental_reverificationMismatch: () => __experimental_reverificationMismatch,
  __experimental_reverificationMismatchResponse: () => __experimental_reverificationMismatchResponse
});
module.exports = __toCommonJS(authorization_errors_exports);
var __experimental_reverificationMismatch = (missingConfig) => ({
  clerk_error: {
    type: "forbidden",
    reason: "reverification-mismatch",
    metadata: {
      reverification: missingConfig
    }
  }
});
var __experimental_reverificationMismatchResponse = (...args) => new Response(JSON.stringify(__experimental_reverificationMismatch(...args)), {
  status: 403
});
var __experimental_isReverificationHint = (result) => {
  var _a, _b;
  return result && typeof result === "object" && "clerk_error" in result && ((_a = result.clerk_error) == null ? void 0 : _a.type) === "forbidden" && ((_b = result.clerk_error) == null ? void 0 : _b.reason) === "reverification-mismatch";
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  __experimental_isReverificationHint,
  __experimental_reverificationMismatch,
  __experimental_reverificationMismatchResponse
});
